crates/cast/src/cmd/storage.rs

```diff
@@ -160,10 +164,15 @@ impl StorageArgs {
         let mut project = etherscan_project(metadata, root_path)?;
         add_storage_layout_output(&mut project);
 
-        project.compiler = if auto_detect {
+        // Override solc version if provided
+         project.compiler = {
+            if let Some(solc_req) = self.solc_version {
+            SolcCompiler::Specific(Solc::find_or_install(&solc_req)?)
+            } else if auto_detect {
             SolcCompiler::AutoDetect
-        } else {
+            } else {
             SolcCompiler::Specific(Solc::find_or_install(&version)?)
+            }
         };
```

Comment:
this isn't formatted correctly, just run `cargo +nightly fmt --all` and it will fix it.